subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecActionFactory.java

```diff
@@ -113,7 +113,12 @@ public ExecAction newDecoratedExecAction() {
 
     @Override
     public ExecAction newExecAction() {
-        return new DefaultExecAction(execHandleFactory.newExecHandleBuilder());
+        return new DefaultExecAction(newExecSpec(), execHandleFactory.newExecHandleBuilder());
+    }
+
+    private ExecSpec newExecSpec() {
+        // In some scopes injection doesn't work, so we inject parameters manually
+        return objectFactory.newInstance(DefaultExecSpec.class, objectFactory, fileResolver);
```

Comment:
❓ Why not a direct constructor call?

Reply:
Since this is object that is instantiated via `objectFactory/instantiator` in other cases, I think we should also prefer it that way here. 
Also in provider-api branch this is an abstract class, so it will be instantiated that way anyway.